src/psyclone/psyir/nodes/codeblock.py

```diff
@@ -38,11 +38,12 @@
 
 ''' This module contains the CodeBlock node implementation.'''
 
-from psyclone.psyir.nodes.node import Node
 from enum import Enum
+from psyclone.psyir.nodes.statement import Statement
+from psyclone.psyir.nodes.datanode import DataNode
 
 
-class CodeBlock(Node):
+class CodeBlock(Statement, DataNode):
```

Comment:
In inherits for both because for the child validation a CodeBlock can be found in the place of both, a Statement or a DataNode. Being child of Node wouldn't be an option. The alternative would be to create 2 classes: `CodeBlockStatement(Statement)` and `CodeBlockDataNode(DataNode)`.

I remember you also had to differentiate them in the back-end and end-up using the Structure attribute and enumeration inside the class. Maybe going to the 2 different classes system will be a better solution altogether?

I will mention something along this lines in the developer guide.

Reply:
Yes I meant developers guide when I said reference guide :-)

Reply:
Sorry if I missed it but I could not see any text talking about CodeBlock(Statement, DataNode) vs CodeBlockStatement(Statement) and CodeBlockDataNode(DataNode) in the developers guide.

Reply:
If it is a better solution then we should make an issue and reference it. If it is not clear then it is fine to discuss in the developers guide.

Reply:
> I will mention something along this lines in the developer guide.

When I wrote this comment in future tense I thought that if I didn't do it immediately I would have a 90% change to forget about it. And I didn't do it immediately :)

I think of it more as an alternative strategy, so I just changed a bit the CodeBlock section of the developer guide to explain this. (I believe it had some exceptions regarding Directives that are not true anymore, so I also changed that part.)